src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -481,9 +495,10 @@ private static Conversion ToConversion(OverloadResolutionResult<MethodSymbol> re
 
             // NOTE: Delegate type compatibility is important, but is not part of the existence check.
 
-            Debug.Assert(method.ParameterCount == parameterCount + (methodGroup.IsExtensionMethodGroup ? 1 : 0));
+            bool isExtensionMethod = methodGroup.IsExtensionMethodGroup && method.ContainingSymbol is not NamedTypeSymbol { IsExtension: true };
```

Comment:
>method.ContainingSymbol is not NamedTypeSymbol { IsExtension: true }; [](http://example.com/codeflow?start=75&length=69)

nit: consider `!method.GetIsNewExtensionMember()`

Reply:
> nit: consider `!method.GetIsNewExtensionMember()`

I'll follow up in a different PR